lib/rules/function-calc-no-unspaced-operator/index.cjs

```diff
@@ -200,82 +100,85 @@ const rule = (primary, _secondaryOptions, context) => {
+						const [, , , , { unit }] = token;
+						const operatorChar = unit.at(-1);
+						const newUnit = unit.slice(0, -1);
 
+						if (complainToFirstOperand(operatorChar, () => cssTokenizer.mutateUnit(token, newUnit))) {
 							return;
 						}
 					}
 
 					if (cssTokenizer.isTokenIdent(token)) {
-						const { value: tokenValue } = token[4];
-						const operatorChar = tokenValue.slice(-1);
-
-						if (operatorChar && OPERATOR_REGEX.test(operatorChar)) {
-							fixOperationWithoutOperator(operation, token, operatorChar);
-
-							if (context.fix) {
-								needsFix = true;
-
-								node.value.splice(
-									node.value.indexOf(operation.firstOperand) + 1,
-									0,
-									operation.operator,
-								);
-
-								cssTokenizer.mutateIdent(token, tokenValue.slice(0, -1));
-							}
+						// E.g. 'id+' → ['id', '+']
+						const [, , , , { value: tokenValue }] = token;
+						const operatorChar = tokenValue.at(-1);
+						const newTokenValue = tokenValue.slice(0, -1);
 
+						if (complainToFirstOperand(operatorChar, () => cssTokenizer.mutateIdent(token, newTokenValue))) {
 							return;
 						}
 					}
 				}
 
 				if (cssParserAlgorithms.isTokenNode(operation.secondOperand) && cssTokenizer.isTokenNumeric(operation.secondOperand.value)) {
 					const token = operation.secondOperand.value;
-					const { signCharacter: operatorChar } = token[4];
+					const [, , startPos, , { signCharacter: operatorChar }] = token;
 
 					if (operatorChar && OPERATOR_REGEX.test(operatorChar)) {
-						operation.operatorToken = [
-							cssTokenizer.TokenType.Delim,
-							operatorChar,
-							token[2],
-							token[3],
-							{ value: operatorChar },
-						];
-						operation.operator = new cssParserAlgorithms.TokenNode(operation.operatorToken);
-
-						if (context.fix) {
-							needsFix = true;
-
-							node.value.splice(node.value.indexOf(operation.secondOperand), 0, operation.operator);
-
-							token[4].signCharacter = undefined;
-							token[1] = token[1].slice(1);
-						}
+						operation.completeMissingOperator(operatorChar, startPos, 'prepend');
+
+						complain(
+							messages.expectedAfter(operation.operatorChar),
+							decl,
+							valueIndex + operation.operatorCharPosition,
+							operation.operatorChar,
+							() => {
+								operation.insertOperatorBeforeSecondOperand(node);
+
+								// Remove an operator character from the second operand token
+								token[4].signCharacter = undefined;
+								token[1] = token[1].slice(1);
+
+								fixDeclarationValue();
+							},
+						);
 					}
 				}
 			}
 
 			/**
-			 * @param {CompleteOperation} operation
-			 * @param {CompleteOperation["before"] | CompleteOperation["after"]} whitespaceNodes
+			 * @param {Operation} operation
+			 * @param {Operation["before"] | Operation["after"]} whitespaceNodes
 			 * @param {'before' | 'after'} position
 			 */
 			function checkOperandWhitespace(operation, whitespaceNodes, position) {
```

Comment:
nit
refactor

Since you have `operation` and `position`, you don't need to pass `whitespaceNodes` anymore.
i.e. `const whitespaceNodes = operation[position];`

Reply:
Nice catch! Fixed via c40a3da88.